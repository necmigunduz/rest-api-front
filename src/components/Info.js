import PropTypes from 'prop-types';
import { icons, measurementUnits } from '../filters/filters';

const MainInfo = (props) => {
  const {
    unit, total,
  } = props;

  return (
    <div className="main-info">
      <p className="big m-b-10 color-dark-gray">{total}</p>
      <i className={`${icons[unit]} m-big color-gray`} />
      <p className="small color-gray">
        {unit}
        <span className="x-small">
          /
          {measurementUnits[unit]}
        </span>
      </p>
    </div>
  );
};

MainInfo.propTypes = {
  unit: PropTypes.string.isRequired,
  total: PropTypes.number.isRequired,
};

export default MainInfo;
