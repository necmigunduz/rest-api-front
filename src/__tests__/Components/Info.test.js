import renderer from 'react-test-renderer';
import Info from '../../components/Info';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'www.instance.com/test',
  }),
}));

describe('Info', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Info
      unit="1"
      measurements={[{}]}
      getMeasurementsByDate={() => {}}
      reduceMethod={() => {}}
      selectedDate={{}}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
