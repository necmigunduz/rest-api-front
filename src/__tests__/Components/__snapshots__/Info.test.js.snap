// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<a
  className="icon background-dark-blue color-white"
  href=""
  onClick={[Function]}
>
  <div>
    <div
      className=" big"
    />
    <div
      className="small m-top-10"
    >
      
    </div>
  </div>
</a>
`;

exports[`Info renders correctly 1`] = `
<div
  className="main-info"
>
  <p
    className="big m-b-10 color-dark-gray"
  />
  <i
    className="undefined x-small color-gray"
  />
  <p
    className="small color-gray"
  >
    1
  </p>
  <p
    className="x-small color-gray"
  />
</div>
`;
